### PR TITLE
Expose --cluster-manager=local for spark-run command.

### PR DESCRIPTION
**Pull request by jsleight**
## Description
Followup from the service_configuration_lib PR to expose the local cluster manager for spark-run in MLCOMPUTE-435.

## Testing done
make test

### CODE REVIEW

File: paasta_tools/cli/cmds/spark_run.py [hunk 1]
```diff
@@ -790,36 +790,15 @@ def configure_and_run_docker_container(
     )
 
     if cluster_manager == CLUSTER_MANAGER_MESOS:
-        volumes = (
-            spark_conf.get("spark.mesos.executor.docker.volumes", "").split(",")
-            if spark_conf.get("spark.mesos.executor.docker.volumes", "") != ""
-            else []
-        )
+        volumes = get_volumes_from_spark_mesos_configs(spark_conf)
     elif cluster_manager == CLUSTER_MANAGER_K8S:
```
Comment (88manpreet) [inline paasta_tools/cli/cmds/spark_run.py:794]:
```suggestion
    elif cluster_manager in [CLUSTER_MANAGER_K8S, CLUSTER_MANAGER_LOCAL]:
```

File: paasta_tools/cli/cmds/spark_run.py [hunk 2]
```diff
@@ -790,36 +790,15 @@ def configure_and_run_docker_container(
...
-            )
-            host_path = spark_conf.get(
-                f"spark.kubernetes.executor.volumes.hostPath.{volume_name}.options.path"
-            )
-            volumes.append(f"{host_path}:{container_path}:{read_only}")
+        volumes = get_volumes_from_spark_k8s_configs(spark_conf)
+    elif cluster_manager == CLUSTER_MANAGER_LOCAL:
+        # service_configuration_lib puts volumes into the k8s
+        # configs for local mode
+        volumes = get_volumes_from_spark_k8s_configs(spark_conf)
```
Comment (88manpreet) [inline paasta_tools/cli/cmds/spark_run.py:799]:
```suggestion
```

**Review by 88manpreet — APPROVED** [review]
lgtm

Please add the ticket id in description. A one line description and test details.
I would still suggest to add some basic manual tests.

File: paasta_tools/cli/cmds/spark_run.py [hunk 3]
```diff
@@ -790,36 +790,13 @@ def configure_and_run_docker_container(
...
-            )
-            container_path = spark_conf.get(
-                f"spark.kubernetes.executor.volumes.hostPath.{volume_name}.mount.path"
-            )
-            host_path = spark_conf.get(
-                f"spark.kubernetes.executor.volumes.hostPath.{volume_name}.options.path"
-            )
-            volumes.append(f"{host_path}:{container_path}:{read_only}")
+        volumes = get_volumes_from_spark_mesos_configs(spark_conf)
+    elif cluster_manager in [CLUSTER_MANAGER_K8S, CLUSTER_MANAGER_LOCAL]:
```
Comment (nemacysts) [inline paasta_tools/cli/cmds/spark_run.py:794]: super-nit:
```suggestion
    elif cluster_manager in {CLUSTER_MANAGER_K8S, CLUSTER_MANAGER_LOCAL}:
```

**Pull request merged**